test/gtest/uct/test_peer_failure.cc

```diff
@@ -56,10 +56,25 @@ class test_uct_peer_failure : public uct_test {
         return UCS_OK;
     }
 
+    typedef struct {
+        uct_pending_req_t    uct;
+        uct_ep_h             ep;
+    } pending_send_request_t;
+
     static ucs_status_t pending_cb(uct_pending_req_t *self)
     {
-        m_req_count++;
-        return UCS_OK;
+        const uint64_t send_data    = 0;
+        pending_send_request_t *req = ucs_container_of(self,
+                                                       pending_send_request_t,
+                                                       uct);
+
+        ucs_status_t status;
+        do {
+            status = uct_ep_am_short(req->ep, 0, 0, &send_data,
+                                     sizeof(send_data));
+        } while (status == UCS_OK);
+
+        return status;
```

Comment:
how does it work? after killing receiver send op should fail, isn't it?

Reply:
it does not fail for RC_MLX5, it's still successful - we need to do `progress()` to get the error
so, to not remove UCT pending request, the test does AM Short operations to get error or fill the resources

Reply:
ic, thanks!

Reply:
> ic, thanks!

done